src/connection.rs

```diff
@@ -74,7 +74,63 @@ impl Connection {
     ///
     /// # Example
     /// ```
-    /// todo!()
+    /// use zbus::{fdo::DBusProxy, MessageType, MatchRule, Address};
```

Comment:
This test should be a [CI test](https://doc.rust-lang.org/book/ch11-03-test-organization.html#the-tests-directory). 
What you are doing is trying to fit a CI test in a doc-test.
From the rust crate perspective, there are doc-tests (check if the doc example sttill does what it says it does), unit tests `#[test]` annotated, these test functions, mostly I believe. These live in the source code itself or are a module or if there are many, a mudule directory `atspi/src/tests/`, and lastly are integration tests (CI). All test- types are tested using cargo test.
 
The CI test is external to the crate (see the link above for more) but associated nevertheless as it lives in `atspi/tests/`. Cargp will compile each file there as separate program and run it as test.

This avoids lots of code here in connection and, more importantly, really links `atspi` as our users would.

Reply:
I disagree, this is exactly how you should use the API, and it's a good example. We just need a way to not display the vast majority of the function due to the overhead of running a command through the shell.

We should absolutely have a doc test for something like this.

But we should *also* automate some CI tests which test different kinds of tests.

Reply:
> I disagree, this is exactly how you should use the API,

Not sure I follow you. What exactly do you disagree on?

> and it's a good example. 

I am not opposing the example, quite the opposite. And if a simple example can double as unit test, that is perfect.

> We just need a way to not display the vast majority of the function due to the overhead of running a command through the shell.

Well you are using the hash `#` there, it should work, but may need a preceding space. 
You can check what it brews by `cargo doc --open`.

> 
> We should absolutely have a doc test for something like this.
> 
I want an example, of course - and it should be doc-tested, but I am not sure the test you are doing here is a good fit. if you want integragion tests, tests that mimics our users and actually links to our library and receive something over a bus, then rusts [integration testing](https://doc.rust-lang.org/book/ch11-03-test-organization.html#integration-tests) is perfect.

Imagine you want 57 tests, for every ATSPI end type one and some for Cache and some for the LitenerEvents, you do not want them here.

> But we should _also_ automate some CI tests which test different kinds of tests.

Reply:
I'm saying that this is a good doc test, it just has a lot of boilerplate. Showing how to use the event_steam() return type is perfect here, and it's what I would expect from other libraries. Even if the test itself is quite long due to the requirement of sending out a signal afterwards.